include/nfc/ndef/msg_parser.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef NFC_NDEF_MSG_PARSER_H_
+#define NFC_NDEF_MSG_PARSER_H_
+
+/**
+ * @file
+ * @defgroup nfc_ndef_msg_parser Parser for NDEF messages
+ * @{
+ * @brief Parser for NFC NDEF messages and records.
+ */
+
+#include <stdint.h>
+#include <zephyr/types.h>
+#include <nfc/ndef/record_parser.h>
+#include <nfc/ndef/nfc_ndef_msg.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Memory allocated for a one-record message.
+ */
+struct nfc_ndef_msg_parser_msg_1 {
+	struct nfc_ndef_msg_desc msg_desc;
+	struct nfc_ndef_record_desc *record_desc_array[1];
+	struct nfc_ndef_bin_payload_desc bin_pay_desc[1];
+	struct nfc_ndef_record_desc rec_desc[1];
+};
+
+/** @brief Memory allocated for a two-record message.
+ */
+struct nfc_ndef_msg_parser_msg_2 {
+	struct nfc_ndef_msg_desc msg_desc;
+	struct nfc_ndef_record_desc *record_desc_array[2];
+	struct nfc_ndef_bin_payload_desc bin_pay_desc[2];
+	struct nfc_ndef_record_desc rec_desc[2];
+};
+
+/** @brief Amount of memory that is required per record in addition to the
+ *         memory allocated for the message descriptor.
+ */
+#define NFC_NDEF_MSG_PARSER_DELTA (sizeof(struct nfc_ndef_msg_parser_msg_2) - \
+				   sizeof(struct nfc_ndef_msg_parser_msg_1))
+
+/** @brief Calculate the memory size required for holding the
+ *         description of a message that consists of a certain number of
+ *         NDEF records.
+ *
+ *  @param[in] max_count_of_records Maximum number of records to hold.
+ */
+#define NFC_NDEF_PARSER_REQIRED_MEMO_SIZE_CALC(max_count_of_records)                    \
+	((u32_t)(max_count_of_records) <= 1) ?                                          \
+	(sizeof(struct nfc_ndef_msg_parser_msg_1) * (uint32_t)(max_count_of_records)) : \
+	(sizeof(struct nfc_ndef_msg_parser_msg_2) + ((NFC_NDEF_MSG_PARSER_DELTA) * ((uint32_t)(max_count_of_records) - 1)))
```

Comment:
Perhaps I'm missing something, but as I understand it, for messages with 2 or more records the value returned by this macro will be larger by `NFC_NDEF_MSG_PARSER_DELTA` than actually needed. For a two-record message it should be just `sizeof(struct nfc_ndef_msg_parser_msg_2)`, shouldn't it?
Why not define this macro as:
```
#define NFC_NDEF_PARSER_REQIRED_MEMO_SIZE_CALC(max_count_of_records) \
	(sizeof(struct nfc_ndef_msg_parser_msg_1) + \
	 ((NFC_NDEF_MSG_PARSER_DELTA) * ((uint32_t)(max_count_of_records) - 1)))
```

Reply:
Ok good idea!